src/D2L.Bmx/MessageWriter.cs

```diff
@@ -41,13 +48,42 @@ void IConsoleWriter.WriteUpdateMessage( string text ) {
 		Console.Error.WriteLine();
 	}
 
-	void IConsoleWriter.WriteWarning( string text ) {
+	void IMessageWriter.WriteWarning( string text ) {
 		string color = _noColor ? "default" : "yellow";
 		_ansiConsole.MarkupLine( $"[{color}]{text}[/]" );
+		WriteToFile( $"[WARNING] {text}" );
 	}
 
-	void IConsoleWriter.WriteError( string text ) {
+	void IMessageWriter.WriteError( string text ) {
 		string color = _noColor ? "default" : "red";
 		_ansiConsole.MarkupLine( $"[{color}]{text}[/]" );
+		WriteToFile( $"[ERROR] {text}" );
+	}
+
+	public void WriteToFile( string text ) {
+		try {
+			string? directory = Path.GetDirectoryName( _logFilePath );
+			if( string.IsNullOrEmpty( directory ) ) {
+				Console.WriteLine( "Unable to determine log file directory." );
+				return;
+			}
+
+			var fileOptions = new FileStreamOptions {
+				Mode = FileMode.Append,
+				Access = FileAccess.Write,
+				Share = FileShare.ReadWrite
+			};
+
+			if( !OperatingSystem.IsWindows() ) {
+				fileOptions.UnixCreateMode = UnixFileMode.UserRead | UnixFileMode.UserWrite;
+			}
+
+			using var stream = new FileStream( _logFilePath, fileOptions );
+			using var writer = new StreamWriter( stream );
+			writer.WriteLine( $"{DateTime.UtcNow:yyyy-MM-dd HH:mm:ss} {text}" );
+			writer.Flush();
+		} catch( Exception ex ) {
+			Console.WriteLine( $"Error writing to log file: {ex.Message}" );
```

Comment:
```suggestion
		string color = _noColor ? "default" : "red";
		_ansiConsole.MarkupLine( $"[{color}] Error writing to log file:  {ex.Message}[/]" );
```


MessageWriter can be invoked in ``bmx print`` or ``write``, in there we can't write misc. text to stdout.

Because we typically use BMX like: ``bmx print | Invoke-Expression`` etc..., we need to make sure we only [pipe](https://www.geeksforgeeks.org/linux-unix/piping-in-unix-or-linux/) text like 
```
$env:AWS_SESSION_TOKEN=123
$env:AWS_ACCESS_KEY_ID=123
$env:AWS_SECRET_ACCESS_KEY=123
```
to the next program.

So we have to write any text intended for the human user elsewhere (to stderr), it will still be visible to the user's terminal but won't get piped.

Reply:
You may want to double check if that code suggestion I provided works, just copy pasted it from above, may have got syntax wrong